tsdb/wlog/watcher.go

```diff
@@ -685,14 +685,12 @@ func (w *Watcher) readSegmentForGC(r *LiveReader, segmentNum int, _ bool) error
 			}
 			w.writer.UpdateSeriesSegment(series, segmentNum)
 
-		// Ignore these; we're only interested in series.
-		case record.Samples:
-		case record.Exemplars:
-		case record.Tombstones:
-
-		default:
+		case record.Unknown:
 			// Could be corruption, or reading from a WAL from a newer Prometheus.
```

Comment:
I didn’t grasp this comment from the initial reading, does it mean?
```suggestion
			// Could be corruption, or version incompatibility between the WAL and Prometheus.
```

Reply:
I would prefer to keep the original comment personally, since it's from before this PR. To me it's also perfectly understandable at least.